### DISCUSSION TITLE
GithHub CI with React: Push to another repo

### DISCUSSION THREAD

**Discussion by federico-arona** [community/community#24298] [Apps, API and Webhooks]
<p>I’ve set a GitHub action that make a build of my React application.</p>
<p>I need that build to be pushed to another repo that I’m using to keep track of the builds. This is the action that is actually running:</p>
<pre><code class="lang-auto">on:
  push:
    branches: [master]

jobs:
  build:
    name: create-package
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [14]

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        name: Use Node.js 14
        with:
          node-version: ${{ matrix.node-version }}
      - run: npm ci

      - run: npm run build --if-present
        env:
          CI: false

  copy:
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Copy to another repo
        uses: andstor/copycat-action@v3
        with:
          personal_token: ${{ secrets.API_TOKEN_GITHUB }}
          src_path: origin-folder/build
          dst_path: /.
          dst_owner: federico-arona
          dst_repo_name: test-build
          dst_branch: main
</code></pre>
<p>By the way when the action run the copy job it fails with the following message:</p>
<p><code>cp: can't stat 'origin-folder/build': No such file or directory</code></p>
<p>What am I doing wrong?</p>
<p>Note that when the react-build scripts start, this is the directory it is running from:</p>
<pre><code class="lang-auto">origin-repo@0.1.0 build /home/runner/work/origin-folder/origin-folder
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>It looks like you’re trying to copy files created by the <code>build</code> job in the <code>copy</code> job. That won’t work, each job gets a fresh runner with an empty file system.</p>
<p>Depending on what fits your needs better, you could either:</p>
<ul>
<li>Build and copy in one job</li>
<li><a href="https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts#passing-data-between-jobs-in-a-workflow">Pass the build result between the jobs as an artifact</a></li>
</ul>

**Comment by federico-arona** [comment]
<p>Thanks airtower-luna for your answer.</p>
<p>I’ve updated the action to this:</p>
<pre><code class="lang-auto">on:
  push:
    branches: [master]

jobs:
  build:
    name: create-package
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [14]

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        name: Use Node.js 14
        with:
          node-version: ${{ matrix.node-version }}
      - run: npm ci

      - run: npm run build --if-present
        env:
          CI: false
      - run: ls
      - name: Copy to another repo
        uses: andstor/copycat-action@v3
        with:
          personal_token: ${{ secrets.API_TOKEN_GITHUB }}
          src_path: build
          dst_path: /.
          dst_owner: federico-arona
          dst_repo_name: test-build
          dst_branch: main

      
</code></pre>
<p>Note the added “ls” command that is giving this output:</p>
<pre><code class="lang-auto">Run ls

README.md
build
node_modules
package-lock.json
package.json
public
src
</code></pre>
<p>By the way the copy command still fails. Any idea?</p>

**Comment by airtower-luna** [comment]
<p>Not really, except that <code>ls</code> showing that the <code>build</code> file or directory exists probably means the problem is with the <code>andstor/copycat-action</code> action. I can’t say if there’s an issue with your config or the action itself.</p>

**Comment by federico-arona** [comment]
<p>Thanks airtower, usign another copy action solved the issue:</p>
<pre><code class="lang-auto">on:
  push:
    branches: [master]

jobs:
  build:
    name: create-package
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [14]

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        name: Use Node.js ${{ matrix.node-version }}
        with:
          node-version: ${{ matrix.node-version }}
      #- name: Install dependencies
      - run: rm -rf node_modules/
      - run: rm -rf build
      - run: npm ci
      - run: npm run build
        env:
          CI: false
      - run: ls
      - name: Copy to another repo
        uses: dmnemec/copy_file_to_another_repo_action@main
        env:
          API_TOKEN_GITHUB: ${{ secrets.API_TOKEN_GITHUB }}
        with:
          source_file: "build"
          destination_repo: "federico-arona/test-build"
          destination_folder: "/."
          user_email: "example@email.com"
          user_name: "example"
          commit_message: "A custom message for the commit"
</code></pre>
<p>Anyway I’m marking your previous answer as accepted because you clarified that each job has its own runner and an empty file system and since I was using two different jobs the problem was also related to that.</p>
<p>Thanks for your support.</p>
